[C#] Add classes exercise (#1374)
exercism/v3: languages/csharp/exercises/concept/classes/.docs/instructions.md

> In this exercise you'll be playing around with a remote controlled car, that you've finally saved enough money for to buy.

Reviewer: ```suggestion
In this exercise you'll be playing around with a remote controlled car, which you've finally saved enough money for to buy.
```

In English, if there is a comma the word after is always "which". If there's not a comma, it's always "that". The difference is whether you need the extra clause to understand or not. In this case you don't need the extra clause, so it should be ", which". (Apologise if you already know all this and just made a typo 😄 )

Author: I didn't know! Thanks for explaining.

> Each time you drive the car using the remote control, it covers 20 meters and drains one percent of the battery (which starts at 100%).

Reviewer: ```suggestion
Cars start with full (100%) batteries. Each time you drive the car using the remote control, it covers 20 meters and drains one percent of the battery.
```

The remote controlled car has a fancy LED display that shows two bits of information:

- The total distance it has driven, displayed as: `"Driven <METERS> meters"`.
- The remaining battery charge, displayed as: `"Battery at <PERCENTAGE>%"`.

If the battery is at 0%, you can't drive the car anymore and the battery display will show `"Battery empty"`.

> You have six tasks, each of which will work with remote controller car instances.

Reviewer 2: ```suggestion
You have six tasks, each of which will work with remote controlled car instances.
```

> ### 1. Buy a brand-new remote controller car

Reviewer 2: ```suggestion
### 1. Buy a brand-new remote controlled car
```

> Implement the (_static_) `Buy()` method to return a brand-new remote controlled car instance:

Author: I'm also unsure if I like omitting the class name. This will be a bit more problematic if there are multiple classes. Another option would be to do:

```suggestion
Implement the (_static_) `RemoteControlCar.Buy()` method to return a brand-new remote controlled car instance:
```

> Implement the (_static_) `Buy()` method to return a brand-new remote controlled car instance:

Author: Or:

```suggestion
Implement the (_static_) `Buy()` method on the `RemoteControlCar` class to return a brand-new remote controlled car instance:
```

> Implement the (_static_) `Buy()` method to return a brand-new remote controlled car instance:

Reviewer: I'm ok with `Implement the (_static_) `RemoteControlCar.Buy()` method`.

> Implement the (_static_) `Buy()` method to return a brand-new remote controlled car instance:

Author: Great

```csharp
RemoteControlCar car = RemoteControlCar.Buy();
```

### 2. Display the distance driven

> Implement the `RemoteControlCar.DistanceDisplay()` method to return the distance as displayed on the LED display:

Reviewer: ```suggestion
Implement the `DistanceDisplay()` method to return the distance as displayed on the LED display:
```

I find it so confusing that both class and instance methods use the same structure in these descriptions. Maybe we could standardise so if it's an instance method, we don't show the class name before it?

Author: Hmmm, I see. What if we would do:

> Implement the `DistanceDisplay()` method to return the distance as displayed on the LED display:

vs.

> Implement the static `DistanceDisplay()` method to return the distance as displayed on the LED display:

To make static the exception?

Reviewer: Yeah, maybe make static bold (* *)?

Author: That could work. Let me update

```csharp
var car = RemoteControlCar.Buy();
car.DistanceDisplay();
// => "Driven 0 meters"
```

### 3. Display the battery percentage

Implement the `RemoteControlCar.BatteryDisplay()` method to return the distance as displayed on the LED display:

```csharp
var car = RemoteControlCar.Buy();
car.BatteryDisplay();
// => "Battery at 100%"
```

### 4. Update the number of meters driven when driving

Implement the `RemoteControlCar.Drive()` method that updates the number of meters driven:

```csharp
var car = RemoteControlCar.Buy();
car.Drive();
car.Drive();
car.DistanceDisplay();
// => "Driven 40 meters"
```

### 5. Update the battery percentage when driving

Update the `RemoteControlCar.Drive()` method to update the battery percentage:

```csharp
var car = RemoteControlCar.Buy();
car.Drive();
car.Drive();
car.BatteryDisplay();
// => "Battery at 98%"
```

> ### 6. Don't allowing driving the car when the battery is drained

Reviewer 2: ```suggestion
### 6. Prevent driving when the battery is drained
```
